src/controllers/request.js

```diff
@@ -95,4 +97,99 @@ export default class usersController {
     });
     }
   }
+  static async rejectRequest(req, res) {
+    try{
+        const { requestId } = req.query;
+        const managerId = req.body.id;
+        
+        const managerExists = await models.User.count(
+            { 
+                where: { id : managerId, role : 'manager'}
+            }
+        ).then( (count) => {
+            return (count > 0) ? true: false;
+        })
+        if(managerExists){
+            const request = await models.Request.findAll(
+                {
+                    where: {
+                        id: requestId,
+                        managerId
+                    }
+                }
+            ).then ((request) => {
+                return request;
+            })
+            
+            
+            if(request.length !== 0){
+                let requestStatus = request[0].status;
+
+                if(requestStatus === 'pending'){
+                    const updatedRequest = await models.Request.update(
+                        { status: 'rejected' },
+                        { where : { id: requestId } }
+                    ).then((request)=> {
+                        return request;
+                        
+                    })
+                    if(updatedRequest.length){
+                        return res.status(200).json({
+                            message: 'The request successfully rejected',
+                            requestId,
+                            requestStatus: updatedRequest[0].status
+                        })
+                    }
+                }
+                else if (requestStatus === 'rejected'){
+                    return res.status(200).json({
+                        message: 'The request was rejected before!',
+                        requestId,
+                        requestStatus
+                    })
+                }
+                else if( requestStatus === 'approved'){
+                    let departureDate = request[0].departureDate;
+                    let todayDate = getTodayDate();
+                    const compareDate = (todayDate, departureDate) => {
+                        return (new Date(todayDate) > new Date(departureDate)) ? true : false;
+                    }
+                    const tripStarted = compareDate(todayDate,departureDate);
+                    
+                    if(tripStarted){
+                        res.status(405).json({
+                            message: "Sorry can't reject ! The user is now on trip."
+
+                        })
+                    } else {
+                        const updatedRequest = await models.Request.update(
+                            { status: 'rejected' },
+                            { where : { id: requestId } }
+                        ).then((request)=> {
```

Comment:
There is no reason for using await and then at the same time, please change that.

Reply:
Corrected.